.github/workflows/liquid-repository-cd-nuget-publish.yml

```diff
@@ -9,40 +9,79 @@ on:
       uses: actions/checkout@v2
-    
-    - name: Setup .NET Core SDK
+      with:
+        fetch-depth: 0 # required to eliminate shallow clone warning in Sonarcloud analysis
+
+    # required by sonarcloud scanner
+    - name: (CI/CD) Setup Java JDK
+      uses: actions/setup-java@v1.4.3
+      with:
+        # The Java version to make available on the path. Takes a whole or semver Java version, or 1.x syntax (e.g. 1.8 => Java 8.x). Early access versions can be specified in the form of e.g. 14-ea, 14.0.0-ea, or 14.0.0-ea.28
+        java-version: 11.x
+        
+    - name: (CI/CD) Setup .NET Core SDK
       uses: actions/setup-dotnet@v1.7.2
       with:
         # SDK version to use. Examples: 2.2.104, 3.1, 3.1.x
         dotnet-version: 3.1.x
 
-    - name: Restore dependencies
+    # required by sonarcloud scanner
+    - name: (CI/CD) Setup Sonar Scanner tool
+      run: dotnet tool install --global dotnet-sonarscanner
+
+    - name: (CI/CD) Install Test Reporting Tool
+      run: dotnet tool install --global dotnet-reportgenerator-globaltool
+
+    - name: (CI/CD) Restore dependencies
       run: dotnet restore src/Liquid.Repository/Liquid.Repository.csproj
 
-    - name: Build Project
-      run: dotnet build src/Liquid.Repository/Liquid.Repository.csproj --configuration Release --no-restore
-      
-    - name: Nuget Pack
-      run: dotnet pack --no-build --configuration Release src/Liquid.Repository/Liquid.Repository.csproj --output .
-      
-    - name: PushNuget
-      if: github.event_name == 'push' 
-      run: dotnet nuget push *.nupkg --source https://api.nuget.org/v3/index.json --api-key ${{secrets.PUBLISH_TO_NUGET_ORG}} --skip-duplicate
+
+
+    - name: (CI/CD) Build and Analyze Project
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN_EF }}
+      run: |
+        dotnet sonarscanner begin /k:"Avanade_Liquid.Repository" /o:"avanade-1" /d:sonar.login="${{secrets.SONAR_TOKEN_EF}}" /d:sonar.host.url="https://sonarcloud.io" /d:sonar.cs.vstest.reportsPaths=$GITHUB_WORKSPACE/test/testresults/*.trx /d:sonar.coverageReportPaths=$GITHUB_WORKSPACE/test/coverlet/reports/SonarQube.xml
+        dotnet build src/Liquid.Repository/Liquid.Repository.csproj --configuration Release --no-restore
+        dotnet test test/Liquid.Repository.Mongo.Tests/*Tests.csproj --collect:"XPlat Code Coverage" --logger trx --results-directory $GITHUB_WORKSPACE/test/testresults
+        reportgenerator -reports:$GITHUB_WORKSPACE/test/testresults/**/coverage.cobertura.xml -targetdir:$GITHUB_WORKSPACE/test/coverlet/reports -reporttypes:"SonarQube"
+        dotnet sonarscanner end /d:sonar.login="${{secrets.SONAR_TOKEN_EF}}"
+    - name: (CD) Nuget Pack & Push to Nuget.org
+      if: ${{ github.event_name == 'push' }}
+      run: |
+        dotnet pack --no-build --configuration Release src/Liquid.Messaging.Azure/Liquid.Messaging.Azure.csproj --output .
```

Comment:
This line is incorrect, this must reference the src/Liquid.Repository/Liquid.Repository.csproj project